man/man1/gfln.1
'\" t
.\"     Title: gfln
.\"    Author: [FIXME: author] [see http://docbook.sf.net/el/author]
.\" Generator: DocBook XSL Stylesheets v1.76.1 <http://docbook.sf.net/>
.\"      Date: 19 Mar 2009
.\"    Manual: Gfarm
.\"    Source: Gfarm
.\"  Language: English
.\"
.TH "GFLN" "1" "19 Mar 2009" "Gfarm" "Gfarm"
.\" -----------------------------------------------------------------
.\" * Define some portability stuff
.\" -----------------------------------------------------------------
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.\" http://bugs.debian.org/507673
.\" http://lists.gnu.org/archive/html/groff/2009-02/msg00013.html
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\" -----------------------------------------------------------------
.\" * set default formatting
.\" -----------------------------------------------------------------
.\" disable hyphenation
.nh
.\" disable justification (adjust text to left margin only)
.ad l
.\" -----------------------------------------------------------------
.\" * MAIN CONTENT STARTS HERE *
.\" -----------------------------------------------------------------
.SH "NAME"
gfln \- create a hardlink or a symbolic link
.SH "SYNOPSIS"
.HP \w'\fBgfln\fR\ 'u
\fBgfln\fR [\fIoptions\fR] target link_name
.SH "DESCRIPTION"
.PP

\fBgfln\fR
creates a hardlink
\fIlink_name\fR
linked to
\fItarget\fR\&. When the \-s option is specified, it creates a symbolic link\&.
.SH "OPTIONS"
.PP
\fB\-s\fR
.RS 4
Creates a symbolic link\&.
.RE
.PP
\fB\-?\fR
.RS 4
Displays a list of command options\&.
.RE


ChangeLog
2007-03-31  Noriyuki Soda  <soda@sra.co.jp>

	* RELNOTES
	avoid an expression which may be misinterpreted as a security problem.

2007-03-31  Osamu Tatebe <tatebe@cs.tsukuba.ac.jp>

	* RELNOTES
	correct description

2007-03-30  Tohru Sotoyama <sotoyama@sra.co.jp>

	* package/solaris/build
	change globus root directory

2007-03-30  Noriyuki Soda  <soda@sra.co.jp>

	* ChangeLog LICENSE RELNOTES
	merge changes in the CVS main trunk until gfarm-1.4.1 to the GNEO
	branch
	
	* ChangeLog
	Catched up with the current ChangeLog.

	* ChangeLog
	recover missed log messages in previous commit
	
	* doc/html/en/ref/man5/gfarm.conf.5.html, doc/html/ja/ref/man5/gfarm.conf.5.html man/ja/man5/gfarm.conf.5 man/man5/gfarm.conf.5:
	regen

	* doc/Gfarm-FAQ.en, doc/Gfarm-FAQ.ja, doc/docbook/en/ref/man5/gfarm.conf.5.docbook, doc/docbook/ja/ref/man5/gfarm.conf.5.docbook:
	merge the following changes in the SourceForge Subversion repository:

	Revision: 3674
		  http://gfarm.svn.sourceforge.net/gfarm/?rev=3674&view=rev
	Author:   sotoyama
	Date:     2007-03-29 02:34:39 -0700 (Thu, 29 Mar 2007)
	Log Message:
	add limitation about mixing gsi authentication with sharedsecret
	found in [gfarm-developers:02814]

	Revision: 3676
		  http://gfarm.svn.sourceforge.net/gfarm/?rev=3676&view=rev
	Author:   sotoyama
	Date:     2007-03-29 04:46:11 -0700 (Thu, 29 Mar 2007)
	Log Message:
	add limitation about mixing gsi authentication with sharedsecret
	found in [gfarm-developers:02814] to `auth' statement

	Revision: 3682
		  http://gfarm.svn.sourceforge.net/gfarm/?rev=3682&view=rev
	Author:   n-soda
	Date:     2007-03-29 11:03:31 -0700 (Thu, 29 Mar 2007)
	Log Message:
	swap section order of limitation and example of the "auth" directive,
	and add a paragraph to describe the example.

2007-03-30  yusuke Tanimura <yusuke.tanimura@aist.go.jp>

	* RELNOTES:
	proofreading

2007-03-29  Noriyuki Soda  <soda@sra.co.jp>

	* server/gfsd/gfsd.c:
	merge the follownig change from the SourceForge Subversion repository:

	Revision: 3681
		  http://gfarm.svn.sourceforge.net/gfarm/?rev=3681&view=rev
	Author:   n-soda
	Date:     2007-03-29 07:37:36 -0700 (Thu, 29 Mar 2007)
	Log Message:
	report error status of close(2) in gfs_server_open_local(),
	problem reported by Markus Elfring as:
	http://sourceforge.net/tracker/index.php?func=detail&aid=1687850&group_id=172718&atid=862742

	Also, treat an error at sendmsg(2) as just an error instead of
	fatal one.

	* bench/fsysbench/fsysbench.c:
	merge the follownig change from the SourceForge Subversion repository:

	Revision: 3679
		  http://gfarm.svn.sourceforge.net/gfarm/?rev=3679&view=rev
	Author:   n-soda
	Date:     2007-03-29 05:13:05 -0700 (Thu, 29 Mar 2007)
	Log Message:
	timer_interrupt_handler() shouldn't call printf(3), but write(2),
	because the former is not async-signal-safe.
	c.f.
	https://sourceforge.net/tracker/?func=detail&atid=862742&aid=1687842&group_id=172718
	
	* ftool/gfarm_agent/gfarm_agent.c:
	merge the follownig change from the SourceForge Subversion repository:

	Revision: 3678
		  http://gfarm.svn.sourceforge.net/gfarm/?rev=3678&view=rev
	Author:   n-soda
	Date:     2007-03-29 04:59:54 -0700 (Thu, 29 Mar 2007)
	Log Message:
	sigterm_handler() shouldn't call exit(3), but _exit(2), because
	the former is not async-signal-safe.
	c.f.
	https://sourceforge.net/tracker/?func=detail&atid=862742&aid=1687842&group_id=172718
	
	* server/gfsd/gfsd.c:
	merge the follownig change from the SourceForge Subversion repository:

	Revision: 3675
		  http://gfarm.svn.sourceforge.net/gfarm/?rev=3675&view=rev
	Author:   n-soda
	Date:     2007-03-29 03:30:02 -0700 (Thu, 29 Mar 2007)
	Log Message:
	do not call exit(3) but _exit(2) in accepting_sigterm_handler(),
	because the former is not async-signal-safe.
	problem reported by Markus Elfring in
	http://sourceforge.net/tracker/index.php?func=detail&aid=1687842&group_id=172718&atid=862742

2007-03-29  yusuke Tanimura <yusuke.tanimura@aist.go.jp>

	* RELNOTES, LICENSE:
	update release notes and the copyright display for the version 1.4.1

2007-03-28  Osamu Tatebe <tatebe@cs.tsukuba.ac.jp>

	* ChangeLog:
	Catched up with the current ChangeLog.

2007-03-28  Noriyuki Soda  <soda@sra.co.jp>

	* include/gfarm/gfarm_misc.h, package/freebsd/Makefile,	package/pkgsrc/Makefile, package/redhat/gfarm.spec, package/solaris/build:
	increase version number to 1.4.1

	* include/gfarm/gfarm_error.h, lib/libgfarm/gfarm/error.c, lib/libgfarm/gfarm/gfs_client.c:
	merge the following change from the SourceForge Subversion repository:

	Revision: 3673
		  http://gfarm.svn.sourceforge.net/gfarm/?rev=3673&view=rev
	Author:   n-soda
	Date:     2007-03-28 07:11:37 -0700 (Wed, 28 Mar 2007)
	Log Message:
	make gfsd reconnection feature work against more network errors.
	this fixes a problem reported by Yamamoto Naotaka-san
	in [gfarm-developers:03456] which was caused by ECONNRESET.

	introduce ENETDOWN, EHOSTDOWN and ENETRESET as gfarm native errors
	for this purpose. (although ENETRESET will be nearly never used)

	* gftool/gneo/gneo.c:
	just a portability fix

	* regress/gfs_hook/lib/fsystest/index.FreeBSD.expected, regress/gfs_hook/lib/fsystest/index.Linux.expected, regress/gfs_hook/lib/fsystest/index.NetBSD.expected, regress/gfs_hook/lib/fsystest/index.SunOS.expected:
	results are different on the GNEO branch with gfs_hook


2.5.7/gftool/gfln/gfln.c
/*
 * $Id: gfmv.c 2640 2006-05-29 22:38:29Z soda $
 */

#include <stdio.h>
#include <stdlib.h>
#include <libgen.h>
#include <unistd.h>

#include <gfarm/gfarm.h>

char *program_name = "gfln";

static void
usage(void)
{
	fprintf(stderr, "Usage: %s [-s] target link_name\n", program_name);
	exit(1);
}

int
main(int argc, char **argv)
{
	gfarm_error_t e;
	int c, status = 0;
	int opt_symlink = 0;
	extern int optind;

	if (argc > 0)
		program_name = basename(argv[0]);
	e = gfarm_initialize(&argc, &argv);
	if (e != GFARM_ERR_NO_ERROR) {
		fprintf(stderr, "%s: %s\n", program_name,
		    gfarm_error_string(e));
		exit(1);
	}

	while ((c = getopt(argc, argv, "sh?")) != -1) {
		switch (c) {
		case 's':
			opt_symlink = 1;
			break;
		case 'h':
		case '?':
		default:
			usage();
		}
	}
	argc -= optind;
	argv += optind;
	if (argc != 2)
		usage();

	e = (opt_symlink ? gfs_symlink : gfs_link)(argv[0], argv[1]);
	if (e != GFARM_ERR_NO_ERROR) {
		fprintf(stderr, "%s%s %s %s: %s\n",
		    program_name, opt_symlink ? " -s" : "",
		    argv[0], argv[1], gfarm_error_string(e));
		status = 1;
	}
	e = gfarm_terminate();
	if (e != GFARM_ERR_NO_ERROR) {
		fprintf(stderr, "%s: %s\n", program_name,
		    gfarm_error_string(e));
		status = 1;
	}
	return (status);
}
